### ISSUE TITLE
Lotka Volterra score 

### ISSUE THREAD

**Issue by Justinezgh** [Justinezgh/SBI-Diff-Simulator#15]
This is the posterior on alpha and delta 
``` python
 plt.hist2d(reference_samples[...,0],reference_samples[...,3],50,cmap ='Blues');
```
![image](https://user-images.githubusercontent.com/72011736/167741069-b08de093-2f0d-418f-950c-003850962ced.png)

and this is the score we get from the lotka_volterra model 
``` python 
plt.quiver(theta[...,0], theta[...,3], score[...,0], score[...,3])
```
![image](https://user-images.githubusercontent.com/72011736/167741208-6b69441a-efeb-48a9-bfa7-2ee26a85b86f.png)
the little blue dot is the posterior 

The posterior (blue) and prior (orange):
![image](https://user-images.githubusercontent.com/72011736/167741892-31753840-2a2f-43d9-b309-be90d5dd2455.png)

I tried with `jax.lax.stop_gradient` in `model_trace['theta']['fn'].log_prob(model_trace['theta']['value'])` but it doesn't change anything

**Comment by Justinezgh** [comment]
I tried with 1e-7 * score and got this for observation 6: 
![image](https://user-images.githubusercontent.com/72011736/167966775-566ce847-9974-40e2-908f-b0ad652457e9.png)
![image](https://user-images.githubusercontent.com/72011736/167966782-24fd5216-15ae-4287-8e5a-283aa25fca1e.png)
![image](https://user-images.githubusercontent.com/72011736/167966792-17c2c9d6-96ee-4fdb-b3b1-46341eb61d63.png)

and for observation 1:
![image](https://user-images.githubusercontent.com/72011736/167966841-8bc4edd9-70d0-4c73-96be-a9924cb6f579.png)

![image](https://user-images.githubusercontent.com/72011736/167966810-d169205c-42bc-4e4b-b2e8-97d6e9750c7c.png)
![image](https://user-images.githubusercontent.com/72011736/167966828-08a8e90b-3d8f-4078-8615-c3ca298677a3.png)


and I tried with 1e-6*score on observation 1:
![image](https://user-images.githubusercontent.com/72011736/167967556-06a2a1ea-098f-458b-b601-bb83acf9caf0.png)
![image](https://user-images.githubusercontent.com/72011736/167967583-a5db2ad0-3f4a-430c-b5eb-564c26536345.png)


[notebook](https://colab.research.google.com/drive/1AMFRjK3c3vWT3KqLjOlekhyuNmMfSTJ3?usp=sharing)

**Comment by Justinezgh** [comment]
hmm sorry that was stupid ^^ 
ok so here I trained with nll + 1e-7 score:
![image](https://user-images.githubusercontent.com/72011736/168078834-dc11cf6f-3320-4d0e-9cc5-436f7945e4b3.png)
![image](https://user-images.githubusercontent.com/72011736/168078873-4795f33a-df94-4bee-b004-1d0144dae308.png)

and with  nll + 1e-6 score:
![image](https://user-images.githubusercontent.com/72011736/168080375-0850c388-b463-4e23-b17b-c0558759c5bc.png)
![image](https://user-images.githubusercontent.com/72011736/168080429-5dda152b-1fc5-429e-ad98-b2399cc7add9.png)





[notebook](https://colab.research.google.com/drive/1AMFRjK3c3vWT3KqLjOlekhyuNmMfSTJ3?usp=sharing)

**Comment by aboucaud** [comment]
The loss seems very unstable.. any clue as to why ?

**Comment by Justinezgh** [comment]
Yes I tried gradient clipping but didn't change anything and it's because sometimes gradients are very big
